### DISCUSSION TITLE
Is it possible to debug my script in IDEs such as Pycharm

### DISCUSSION THREAD

**Discussion by TenyaChannel** [3b1b/manim#1701] [Q&A] [answered]
I want to debug my script in Pycharm but I cannot make it. I must run the command "_manimgl <my_script> <my_scene_name>_" which means my script is loaded into the manimgl executable and Pycharm cannot debug it.

I tried to follow instructions in _https://p14jeffwest.blogspot.com/2020/04/how-to-debug-manim-code-with-pycham.html_, but it is for the old version of manim. In the new version, I could not find the _manim.py_ file, and the _manimgl_ module does not have the attribute _main_ so I cannot directly use _manim.py_  in the old version.

I also tried to attach Pdb to Pycharm but I failed either.

I wounder if it is kind of you to provide something like _manim.py_ to enable debugging in IDEs.

**Comment by TenyaChannel** [comment, marked as answer]
update: I found it in manimlib/__main__.py, now I can debug scripts in Pycharm just by changing the command to python <package_path>/manimlib/__main__.py <script_path> <scene>

**Reply by InspireCode** [reply]
I can not find main.py in your provided path.
It really trouble me to debug a manimgl script.
